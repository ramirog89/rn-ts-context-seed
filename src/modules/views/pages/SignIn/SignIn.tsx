import React, { useCallback, useState } from 'react';
import { View, TextInput, Button } from 'react-native';
import { useAuth } from '../../../context/auth';

const SignInScreen = () => {
  const { signIn } = useAuth();

  const [username, setUsername] = useState<string>('');
  const [password, setPassword] = useState<string>('');

  const onSignIn = useCallback(() => {
    signIn(username, password);
  }, [username, password, signIn]);

  return (
    <View>
      <TextInput
        placeholder="Username"
        value={username}
        onChangeText={setUsername}
      />
      <TextInput
        placeholder="Password"
        value={password}
        onChangeText={setPassword}
        secureTextEntry
      />
      <Button title="Sign in" onPress={onSignIn} />
    </View>
  );
}

export default SignInScreen;
