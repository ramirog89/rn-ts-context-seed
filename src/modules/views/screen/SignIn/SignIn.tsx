import React, { useCallback, useState } from 'react';
import { Form, Item, Input, Button, Text, Icon, Container, Content } from 'native-base';

import { useAuth } from '../../../context/auth';

const SignInScreen = () => {
  const { signIn } = useAuth();
  const [username, setUsername] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [isLoading, setLoading] = useState<boolean>(false);
  
  const onSignIn = useCallback(async () => {
    setLoading(true);
    try {
      await signIn({ username, password });
    } catch (e) {
      setLoading(false);
    }
  }, [username, password, signIn]);

  return (
    <Container style={{ backgroundColor: '#546e7a' }}>
      <Content>
        <Form style={{ width: '95%', marginTop: '20%', marginBottom: '7%', }}>
          <Item>
            <Icon name="person" style={{ color: '#FFFFFF' }} />
            <Input
              style={{ color: '#FFFFFF' }}
              placeholderTextColor="#BBBBBB"
              placeholder="Username"
              value={username}
              onChangeText={setUsername}
            />
          </Item>
          <Item>
            <Icon name="key" style={{ color: '#FFFFFF' }} />
            <Input
              style={{ color: '#FFFFFF' }}
              placeholder="Password"
              placeholderTextColor="#BBBBBB"
              value={password}
              onChangeText={setPassword}
              secureTextEntry
            />
          </Item>
        </Form>
        <Button style={{ marginTop: 15, backgroundColor: '#29434e' }} full disabled={isLoading} onPress={onSignIn}>
          <Text>Sign In</Text>
        </Button>
      </Content>
    </Container>
  );
}

export default SignInScreen;
