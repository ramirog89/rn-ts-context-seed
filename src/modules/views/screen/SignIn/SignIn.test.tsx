import React from 'react';

import { mockedRoute } from '../../../../test/setup';
import { render } from '../../../../test/render';

import SignIn from './SignIn';

describe('SignInScreen', () => {
  beforeEach(() => {
    mockedRoute.mockReturnValue('SignIn');
  });

  it('should render', () => {
    const { toJSON } = render(<SignIn />);
    expect(toJSON()).toMatchSnapshot();
  });
});
