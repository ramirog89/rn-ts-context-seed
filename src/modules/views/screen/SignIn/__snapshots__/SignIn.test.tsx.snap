// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreen should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#546e7a",
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          style={
            Array [
              Object {},
              Object {
                "marginBottom": "7%",
                "marginTop": "20%",
                "width": "95%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={false}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginLeft": 15,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                      "paddingRight": 8,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <TextInput
              allowFontScaling={true}
              editable={true}
              onChangeText={[Function]}
              placeholder="Username"
              placeholderTextColor="#BBBBBB"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "top": 1.5,
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
              value=""
            />
          </View>
          <View
            accessible={true}
            focusable={false}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginLeft": 15,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                      "paddingRight": 8,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <TextInput
              allowFontScaling={true}
              editable={true}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#BBBBBB"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "top": 1.5,
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
              value=""
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#29434e",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 0,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginTop": 15,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            Sign In
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
